proof/refine/ARM/Syscall_R.thy

```diff
@@ -1119,7 +1119,7 @@ lemma handleInvocation_corres:
             apply (rule corres_split [OF setThreadState_corres], simp)
               apply (rule corres_splitEE [OF _ performInvocation_corres])
                   apply simp
-                  apply (rule corres_split [OF getThreadState_corres])
+                  apply (rule corres_split [OF getThreadState_corres]; (solves simp)?)
```

Comment:
Would a `lemmas getThreadState_corres' = getThreadState_corres[OF refl]` be worth making to reduce the number of places you need this `solves simp` step?

Reply:
I'd be happy to do that. But if I did it to this rule, should I do it for the others too? What does everyone think?